test/libsolidity/syntaxTests/nameAndTypeResolution/508_large_storage_arrays_combined.sol

```diff
@@ -2,4 +2,3 @@ contract C {
     uint[200][200][2**30][][2**30] x;
 }
 // ----
-// Warning: (17-49): Variable covers a large part of storage and thus makes collisions likely. Either use mappings or dynamic arrays and allow their size to be increased only in small quantities per transaction.
```

Comment:
We could either remove this test. Or write a new test where an error is found.

Perhaps just ` uint[200][200][2**30][2**30] x;`

Reply:
I think the test verifies that a dynamic array "in the middle" noes not cause the error.

Reply:
That actually sounds good!

Reply:
I'm confused: this change does not belong to this PR, does it?

Reply:
@bshastry 
> I'm confused: this change does not belong to this PR, does it?

I would say it does :). The previous warning is gone, because dynamic array size is counted as `1`.

Reply:
@a3d4 But doesn't this change only impact dynamically sized arrays *within* structs i.e., https://github.com/ethereum/solidity/pull/9187/files#diff-aa50ebe623fccb833f2b9b4c214820c9R349 is part of `structureSizeEstimate` which is presumably only called for structs and not top level dynamically sized storage arrays?

Reply:
@bshastry 
>  which is presumably only called for structs and not top level dynamically sized storage arrays

Not quite :). We have:

```
else if (_variable.isStateVariable())
{
	set<StructDefinition const*> structsSeen;
	if (structureSizeEstimate(*_variable.type(), structsSeen) >= bigint(1) << 64)
		m_errorReporter.warning(
. . . .
```

In fact, the next step might be to remove `StaticAnalyzer::structureSizeEstimate()` and add `Types::storageSizeEstimate() ` instead.